cmd/kubelet/app/server.go

```diff
@@ -123,184 +104,54 @@ HTTP server: The kubelet can also listen for HTTP and respond to a simple API
-		KubeClient:                   nil,
-		ManifestURL:                  s.ManifestURL,
-		ManifestURLHeader:            manifestURLHeader,
-		MasterServiceNamespace:       s.MasterServiceNamespace,
-		MaxContainerCount:            int(s.MaxContainerCount),
-		MaxOpenFiles:                 uint64(s.MaxOpenFiles),
-		MaxPerPodContainerCount:      int(s.MaxPerPodContainerCount),
-		MaxPods:                      int(s.MaxPods),
-		NvidiaGPUs:                   int(s.NvidiaGPUs),
-		MinimumGCAge:                 s.MinimumGCAge.Duration,
-		Mounter:                      mounter,
-		NetworkPluginName:            s.NetworkPluginName,
-		NetworkPlugins:               ProbeNetworkPlugins(s.NetworkPluginDir),
-		NodeLabels:                   s.NodeLabels,
-		NodeStatusUpdateFrequency:    s.NodeStatusUpdateFrequency.Duration,
-		NonMasqueradeCIDR:            s.NonMasqueradeCIDR,
-		OOMAdjuster:                  oom.NewOOMAdjuster(),
-		OSInterface:                  kubecontainer.RealOS{},
-		PodCIDR:                      s.PodCIDR,
-		ReconcileCIDR:                s.ReconcileCIDR,
-		PodInfraContainerImage:       s.PodInfraContainerImage,
-		Port:                           uint(s.Port),
-		ReadOnlyPort:                   uint(s.ReadOnlyPort),
-		RegisterNode:                   s.RegisterNode,
-		RegisterSchedulable:            s.RegisterSchedulable,
-		RegistryBurst:                  int(s.RegistryBurst),
-		RegistryPullQPS:                float64(s.RegistryPullQPS),
-		ResolverConfig:                 s.ResolverConfig,
-		Reservation:                    *reservation,
-		KubeletCgroups:                 s.KubeletCgroups,
-		RktPath:                        s.RktPath,
-		RktAPIEndpoint:                 s.RktAPIEndpoint,
-		RktStage1Image:                 s.RktStage1Image,
-		RootDirectory:                  s.RootDirectory,
-		SeccompProfileRoot:             s.SeccompProfileRoot,
-		Runonce:                        s.RunOnce,
-		SerializeImagePulls:            s.SerializeImagePulls,
-		StandaloneMode:                 (len(s.APIServerList) == 0),
-		StreamingConnectionIdleTimeout: s.StreamingConnectionIdleTimeout.Duration,
-		SyncFrequency:                  s.SyncFrequency.Duration,
-		SystemCgroups:                  s.SystemCgroups,
-		TLSOptions:                     tlsOptions,
-		Writer:                         writer,
-		VolumePlugins:                  ProbeVolumePlugins(s.VolumePluginDir),
-		OutOfDiskTransitionFrequency:   s.OutOfDiskTransitionFrequency.Duration,
-		HairpinMode:                    s.HairpinMode,
-		BabysitDaemons:                 s.BabysitDaemons,
-		ExperimentalFlannelOverlay:     s.ExperimentalFlannelOverlay,
-		NodeIP:         net.ParseIP(s.NodeIP),
-		EvictionConfig: evictionConfig,
-		PodsPerCore:    int(s.PodsPerCore),
+		writer = &kubeio.NsenterWriter{}
+	}
+
+	return &kubelet.KubeletDeps{
+		Auth:              nil, // default does not enforce auth[nz]
+		CAdvisorInterface: nil, // launches background processes, not set here
```

Comment:
It was export but also some of the background cleaners.  This method was
split into two parts - initialization that was side effect free, and
initialization that could not be turned back around (goroutines running
with no way to shut down).

On Fri, Aug 12, 2016 at 1:06 PM, Michael Taufen <notifications@github.com>
wrote:

> In cmd/kubelet/app/server.go
> <https://github.com/kubernetes/kubernetes/pull/29216#discussion_r74625309>
> :
>
> > -		TLSOptions:                     tlsOptions,
> > -		Writer:                         writer,
> > -		VolumePlugins:                  ProbeVolumePlugins(s.VolumePluginDir),
> > -		OutOfDiskTransitionFrequency:   s.OutOfDiskTransitionFrequency.Duration,
> > -		HairpinMode:                    s.HairpinMode,
> > -		BabysitDaemons:                 s.BabysitDaemons,
> > -		ExperimentalFlannelOverlay:     s.ExperimentalFlannelOverlay,
> > -		NodeIP:         net.ParseIP(s.NodeIP),
> > -		EvictionConfig: evictionConfig,
> > -		PodsPerCore:    int(s.PodsPerCore),
> > +		writer = &kubeio.NsenterWriter{}
> > +	}
> > +
> > +	return &kubelet.KubeletDeps{
> > +		Auth:              nil, // default does not enforce auth[nz]
> > +		CAdvisorInterface: nil, // launches background processes, not set here
>
> @smarterclayton <https://github.com/smarterclayton> do you know which
> background processes cadvisor.New supposedly started? Looks like you
> added that comment in bb79365#diff-e0160f895346669c2e4495acfd858ea7R331
> <https://github.com/kubernetes/kubernetes/commit/bb79365e2aa0c38673d1e7f31f74d04982965820#diff-e0160f895346669c2e4495acfd858ea7R331>
> .
>
> Were you maybe referring to exportHTTP doing a ListenAndServe in the
> background?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/29216/files/8d7ae7df3d1d481c74855447afad10b110c8360c#r74625309>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_p15HypUvrYjvxhLZYAZaflC2AUZPks5qfKf9gaJpZM4JP-G6>
> .
>